tests/features/custom-order/custom-order.mysql.test.ts

```diff
@@ -0,0 +1,274 @@
+import { Collection, Entity, Enum, Logger, ManyToOne, MikroORM, OneToMany, PrimaryKey, Property, QueryOrder } from '@mikro-orm/core';
+import type { MySqlDriver } from '@mikro-orm/mysql';
+
+type Rating = 'bad' | 'ok' | 'good';
+
+enum Priority {
+  Low = 'low',
+  Medium = 'medium',
+  High = 'high'
+}
+
+// Out of order on purpose
+enum Difficulty {
+  Easy = 2,
+  Hard = 1,
+  Medium = 0
+}
+
+@Entity()
+class User {
+
+  @PrimaryKey()
+  id!: number;
+
+  @Property()
+  name!: string;
+
+  // eslint-disable-next-line @typescript-eslint/no-use-before-define
+  @OneToMany(() => Task, ({ owner }) => owner)
+  tasks = new Collection<Task>(this);
+
+}
+
+@Entity()
+class Task {
+
+  @PrimaryKey()
+  id!: number;
+
+  @Property()
+  label!: string;
+
+  @ManyToOne(() => User, { nullable: true })
+  owner?: User;
+
+  @Enum({
+    items: () => Priority,
+    customOrder: [Priority.Low, Priority.Medium, Priority.High],
+    nullable: true,
+  })
+  priority?: Priority;
+
+  @Property({ customOrder: ['bad', 'ok', 'good'], nullable: true })
+  rating?: Rating;
+
+  @Enum({
+    items: () => Difficulty,
+    customOrder: [Difficulty.Easy, Difficulty.Medium, Difficulty.Hard],
+    nullable: true,
+  })
+  difficulty?: Difficulty;
+
+}
+
+const createWithPriority = (label: string, priority?: Priority) => {
+  const t = new Task();
+  t.label = label;
+  t.priority = priority;
+  return t;
+};
+
+const createWithRating = (label: string, rating?: Rating) => {
+  const t = new Task();
+  t.label = label;
+  t.rating = rating;
+  return t;
+};
+
+const createWithDifficulty = (label: string, difficulty?: Difficulty) => {
+  const t = new Task();
+  t.label = label;
+  t.difficulty = difficulty;
+  return t;
+};
+
+describe('custom order [mysql]', () => {
+
+  let orm: MikroORM<MySqlDriver>;
+
+  beforeAll(async () => {
+    orm = await MikroORM.init<MySqlDriver>({
+      entities: [Task, User],
+      dbName: `mikro_orm_test_custom_order`,
+      type: 'mysql',
+      port: 3307,
+    });
+
+    await orm.getSchemaGenerator().ensureDatabase();
+    await orm.getSchemaGenerator().dropSchema();
+    await orm.getSchemaGenerator().createSchema();
+  });
+  beforeEach(async () => {
+    await orm.em.nativeDelete(Task, {});
+    await orm.em.nativeDelete(User, {});
+  });
+  afterAll(async () => orm.close(true));
+
+  test('query string enum ASC', async () => {
+    const mock = jest.fn();
+    const logger = new Logger(mock, ['query']);
+    Object.assign(orm.config, { logger });
```

Comment:
after you rebase this with latest changes from master, you will need to change this in all the tests to use `mockLogger()` function from `bootstrap.ts`
```suggestion
    const mock = mockLogger(orm, ['query']);
```

(don't forget to remove the import for `Logger`)